content/articles/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/index.md

```diff
@@ -0,0 +1,261 @@
+This is the color of the circle.
+- `cv2.FILLED`:
+This refers to the thickness of -1 pixels. It will fill the circle with the color we specify.
+
+We will repeat the same for the index finger in the line of code: 
+
+`cv2.circle(img, (x2, y2), 15, (255, 0, 0), cv2.FILLED)`. 
+
+### Drawing a line between points 4 and 8
+```python
+        cv2.line(img, (x1, y1), (x2, y2), (255, 0, 0), 3)
+```
+In the above code, we are using the cv2.line function to draw a line between point four of the hand, and point 8.
+- `img`:
+We are passing our image to the function.
+- `(x1, y1), (x2, y2)`:
+We are specifying that the line we will draw will connect point 4 (`(x1, y1)`) which is the tip of the thumb and point 8 (`(x2, y2)`) which is the tip of the index finger.
+- `(255, 0, 0)`:
+This is the color of the line.
+- `3`:
+This is the thickness of the line.
+
+### Finding the distance between points 4 and 8
+```python
+        length = hypot(x2 - x1, y2 - y1)
+```
+In the above code, we are finding the distance between the tip of the thumb and the index finger using hypotenuse. We achieve this by calling the math hypot function and passing the difference between x2 and x1 and the difference between y2 and y1.
+### Converting the hand range to the volume range
+```python                                          
+        vol = np.interp(length, [15, 220], [volMin, volMax])
+        print(vol, length)
+```
+We are calling on the numpy function `np.interp` to convert the hand range to the volume range. The arguments used are:
+- `length`: This is the value we want to convert.
+- `[150 - 220]`: This is the hand range.
+- `[volMin, volMax]`: Giving the range to which we want to convert.
+
+### Setting the master volume
+```python        
+        volume.SetMasterVolumeLevel(vol, None)  
+```
+We are setting the master volume level in accordance with the hand range. We achieve this by passing `vol` which is the value of the hand range we did convert to volume range.
+
+### Displaying the video output used to interact with the user
+```python
+    cv2.imshow('Image', img) 
+```
+The above code shows the real-time video showing the user interacting with the program. i.e The user uses the thumb finger and the index finger to control the volume.
+
+### Terminating the program
+```python
+    if cv2.waitKey(1) & 0xff == ord('q'): 
+        break
+```
+The above code will terminate the program if the user presses the q key.
+### Results
+When the code above has run to completion without any errors, the results will be as those shown in the screenshot below.
+
+![Results](/engineering-education/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/results.gif)
+
+### Conclusion 
+You now have all the skills required to create a hand gesture volume controller. Do your work while listening to your favorite music. You don't have to worry about dirtying your hands. By just a gesture of your hand, you can control the volume level of your music. You

```

Comment:
Remove the `You` at the end of the sentence. You need to proofread your work to avoid unnecessary corrections.